stdlib/public/runtime/ReflectionMirror.cpp

```diff
@@ -1126,4 +1127,28 @@ id swift_reflectionMirror_quickLookObject(OpaqueValue *value, const Metadata *T)
 }
 #endif
 
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERNAL
+const char *swift_keypath_dladdr(void *address) {
+  SymbolInfo info;
+  if (lookupSymbol(address, &info) == 0) {
+    return 0;
+  } else {
+    return info.symbolName.get();
+  }
+}
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERNAL
+const void *swift_keypathSourceString(
+                                      char *name,
+                                      void *(callback)(const char *)
+                                      ) {
+  size_t length = strlen(name);
+  std::string mangledName = keypathSourceString(name, length);
+  if (mangledName == "") {
+    return 0;
+  } else {
+    return callback(mangledName.c_str());
```

Comment:
Ok, I've went back and forth on this, but I think it'd just be easier to call `strdup` here and return that pointer from this function. Then on the Swift side you can call `String(cString:)` from that pointer and just `deallocate()` when you're done.